File: backend/src/main/java/com/example/tracker/service/TecnicoService.java
package com.example.tracker.service;

import com.example.tracker.dto.tecnico.TecnicoCreateDTO;
import com.example.tracker.dto.tecnico.TecnicoResponseDTO;
import java.util.List;
import java.util.Optional;

public interface TecnicoService {

    TecnicoResponseDTO criar(TecnicoCreateDTO dto);

    List<TecnicoResponseDTO> listarTecnicos();

    List<TecnicoResponseDTO> listarSelecionaveis();

    Optional<TecnicoResponseDTO> buscarPorId(Integer id);

    TecnicoResponseDTO atualizar(Integer id, TecnicoCreateDTO dto);

    void deletar(Integer id);
}

File: frontend/src/services/tecnicoService.ts
import { apiFetch } from './api'

export interface TecnicoResponseDTO {
  id: number
  usuarioId: number
  email: string
  nome: string
  cpf: string
  cargo: string
  telefone: string
  latitude: number
  longitude: number
  certificacao: string | null
  estado: string | null
  disponibilidade: string | null
}

export interface TecnicoCreateDTO {
  email: string
  senha: string
  nome: string
  cpf?: string
  cargo?: string
  telefone?: string
  latitude?: number
  longitude?: number
}

export const tecnicoService = {
  listar: () => apiFetch<TecnicoResponseDTO[]>('/tecnicos'),
  buscarPorId: (id: number) => apiFetch<TecnicoResponseDTO>(`/tecnicos/${id}`),
  criar: (dto: TecnicoCreateDTO) => apiFetch<TecnicoResponseDTO>('/tecnicos', {
    method: 'POST',
    body: JSON.stringify(dto),
  }),
  atualizar: (id: number, dto: TecnicoCreateDTO) => apiFetch<TecnicoResponseDTO>(`/tecnicos/${id}`, {
    method: 'PUT',
    body: JSON.stringify(dto),
  }),
  deletar: (id: number) => apiFetch<void>(`/tecnicos/${id}`, {
    method: 'DELETE',
  }),
}